File: benchmarks/json/generate_10k.nim
import std/[times, json, strformat]
var items = newJArray()
for i in 0..<10000:
  items.add(%*{"id": i, "name": fmt"item_{i}", "tags": ["t0","t1","t2","t3","t4"], "value": float(i) * 2.718})
let start = cpuTime()
let result = $items
echo int((cpuTime() - start) * 1_000_000)

File: benchmarks/json/generate_10k.rs
use std::time::Instant;
fn main() {
    let start = Instant::now();
    let mut result = String::with_capacity(1024 * 1024);
    result.push('[');
    for i in 0..10000 {
        if i > 0 { result.push(','); }
        result.push_str(&format!(
            "{{\"id\":{},\"name\":\"item_{}\",\"tags\":[\"t0\",\"t1\",\"t2\",\"t3\",\"t4\"],\"value\":{:.3}}}",
            i, i, i as f64 * 2.718
        ));
    }
    result.push(']');
    println!("{}", start.elapsed().as_micros());
}
